src/toil/lib/aws/iam.py

```diff
@@ -63,8 +89,14 @@ def policy_permissions_allow(given_permissions: Set[str] = set(), required_permi
 
     # We only check actions explicitly allowed on all resources here,
     #TODO: Add a resource parameter to check for actions allowed by resource
+    resource = "*"
+
+    missing_perms = []
 
-    missing_perms = [x for x in required_permissions if not check_permission_allowed(x, list(given_permissions))]
+    for permission in required_permissions:
+        if not check_permission_allowed(permission, given_permissions[resource]['Action']):
+            if given_permissions[resource]['NotAction'] == [] or check_permission_allowed(permission, given_permissions[resource]["NotAction"]):
```

Comment:
This use here makes me want to rename `check_permission_allowed` to something more like `permission_matches_any`, because if we match a NotAction list, then we *aren't* allowed by that list, and so here we're kind of using the function backward and it makes this already-complex code more confusing.